crates/common/consensus/src/execution_engine/mod.rs

```diff
@@ -108,12 +107,33 @@ impl ExecutionEngine {
             .json(&request_body)
             .bearer_auth(self.create_jwt_token()?)
             .build();
-        Ok(self
-            .http_client
+        self.http_client
             .execute(http_post_request?)
             .await?
             .json::<JsonRpcResponse<Vec<String>>>()
-            .await
-            .map(|result| result.result)?)
+            .await?
+            .to_result()
+    }
+
+    pub async fn engine_get_payloadv3(&self, payload_id: B64) -> anyhow::Result<PayloadV3> {
+        let request_body = JsonRpcRequest {
+            id: 1,
+            jsonrpc: "2.0".to_string(),
+            method: "engine_getPayloadV3".to_string(),
+            params: vec![serde_json::json!(payload_id)],
+        };
+        let http_post_request = self
+            .http_client
+            .post(&self.engine_api_url)
+            .json(&request_body)
+            .bearer_auth(self.create_jwt_token()?)
+            .build();
+
+        self.http_client
+            .execute(http_post_request?)
+            .await?
+            .json::<JsonRpcResponse<PayloadV3>>()
+            .await?
+            .to_result()
```

Comment:
```suggestion

    pub fn build build_request(&self, rpc_request: JsonRpcRequest) -> anyhow::Result<Request> {
        self
            .http_client
            .post(&self.engine_api_url)
            .json(&request_body)
            .bearer_auth(self.create_jwt_token()?)
            .build()
        }

    pub async fn engine_get_payloadv3(&self, payload_id: B64) -> anyhow::Result<PayloadV3> {
        let request_body = JsonRpcRequest {
            id: 1,
            jsonrpc: "2.0".to_string(),
            method: "engine_getPayloadV3".to_string(),
            params: vec![serde_json::json!(payload_id)],
        };
        let http_post_request = self.build_request(request_body)?;

        self.http_client
            .execute(http_post_request)
            .await?
            .json::<JsonRpcResponse<PayloadV3>>()
            .await?
            .to_result()
```

can you apply this to your functions above as well to reduce code duplication